docs/das-runner.md

```diff
@@ -0,0 +1,98 @@
+# GitHub Self-Hosted Runner Setup Guide
+
+This guide walks through the steps required to configure a self-hosted GitHub Actions runner for projects in the `singnet` organization using the `github-runner` script.
+
+## Script Overview
+
+The script automates the process of:
+1. Connecting to a remote server (using either password or SSH key authentication).
+2. Fetching the list of GitHub repositories.
+3. Selecting a repository to configure the runner for.
+4. Installing the GitHub Actions runner on the server.
+5. Configuring the runner and creating a `systemd` service to manage it.
+
+## Pre-requisites
+
+Before running the script, ensure the following:
+1. The `gh` CLI is installed and authenticated with GitHub.
+   - You can install the `gh` CLI by following [this link](https://cli.github.com/).
+2. You have SSH access to the server where the runner will be installed.
+3. `jq` is installed on the local machine to parse JSON responses.
+
+### Installing `gh` and `jq`
+
+```bash
+# Install GitHub CLI
+sudo apt install gh
+
+# Install jq
+sudo apt install jq
+```
+
+## Usage Instructions
+
+### 1. Running the Script
+
+Start the script with the following command:
+
+```bash
+make github-runner
+```
+
+### 2. Enter Server Details
+
+The script will prompt you for the following details:
+- **Server IP:** The IP address of the server where the GitHub runner will be installed.
+- **Username:** The username for SSH access to the server.
+
+### 3. Authentication Method
+
+Select the method to authenticate on the server:
+- **1) Password:** Use a password for SSH.
+- **2) Private Key:** Use a private key for SSH.
```

Comment:
What if the private key is protected with a passphrase?

Reply:
In this case, you will be prompted to enter the SSH password. @Pedrobc89